lua/diffview/actions.lua

```diff
@@ -336,19 +336,83 @@ local function diff_copy_target(kind)
   end
 end
 
+---@param target "ours"|"theirs"|"base"|"all"|"none"
+function M.conflict_choose_file(target)
+  return function()
+    local view = lib.get_current_view()
+
+    if view and view:instanceof(StandardView.__get()) then
+      ---@cast view StandardView
+      local main = view.cur_layout:get_main_win()
+      local curfile = main.file
+
+      if main:is_valid() and curfile:is_valid() then
+        local lines = api.nvim_buf_get_lines(curfile.bufnr, 0, -1, false)
+        local conflicts = vcs_utils.parse_conflicts(
+          lines,
+          main.id
+        )
+
+        if conflicts then
+          local content
+          local drift_lines = 0
+          for _, cur in pairs(conflicts) do
+
+            print("cur: "..vim.inspect(cur))
+
+            -- add drift of line numbers
+            cur.first = cur.first + drift_lines
+            cur.last = cur.last + drift_lines
+            cur.ours.first = cur.ours.first + drift_lines
+            cur.ours.last = cur.ours.last + drift_lines
+            cur.theirs.first = cur.theirs.first + drift_lines
+            cur.theirs.last = cur.theirs.last + drift_lines
+
+            print("cur with drift: "..vim.inspect(cur))
+
+            if target == "ours" then content = cur.ours.content
+            elseif target == "theirs" then content = cur.theirs.content
+            elseif target == "base" then content = cur.base.content
+            elseif target == "all" then
+              content = utils.vec_join(
+                cur.ours.content,
+                cur.base.content,
+                cur.theirs.content
+              )
+            end
+
+            api.nvim_buf_set_lines(curfile.bufnr, cur.first - 1, cur.last, false, content or {})
+
+            utils.set_cursor(main.id, unpack({
+              (content and #content or 0) + cur.first - 1,
+              content and content[1] and #content[#content] or 0
+            }))
+
+            local lines_count = 0
+            for _ in pairs(content or {}) do lines_count = lines_count + 1 end
+            drift_lines = lines_count - (cur.last - cur.first + 1)
```

Comment:
I just thought, do I need to move the cursor at all here? Can you think of any edge case where it could lead to an issue if I just let the cursor where it is?

Reply:
There's never a technical reason to move the cursor here. We move the cursor for UX reasons. It's to communicate to the user that "something changed in this buffer, in this location". With this action we potentially change the buffer in multiple locations, but I think it makes sense to place the cursor in place of the last conflict.